app/src/main/java/org/xbmc/kore/ui/sections/file/MediaFileListFragment.java

```diff
@@ -511,16 +512,15 @@ public boolean onMenuItemClick(MenuItem item) {
                                     case R.id.action_play_from_this_item:
                                         mediaQueueFileLocation.clear();
                                         FileLocation fl;
+                                        // start playing the selected one, then queue the rest
+                                        mediaQueueFileLocation.add(loc);
+                                        playMediaFile(loc.file);
```

Comment:
Slight possible bug here. `PlayMediaFile` needs to be the last thing on this block, as it also sends the queued items to Kodi, so `mediaQueueFileLocation` needs to be full before calling it.
Actually i don't think that `PlayMediaFile` should do this, but it's how it's implemented right now, so to avoid a possible bug this needs to be left as the last line before the return.